File: src/main/java/cn/don9cn/blog/action/system/log/SysOperaLogAction.java
package cn.don9cn.blog.action.system.log;

import cn.don9cn.blog.action.BaseAction;
import cn.don9cn.blog.model.system.log.SysOperaLog;
import cn.don9cn.blog.support.daohelper.core.PageResult;
import cn.don9cn.blog.service.system.log.interf.SysOperaLogService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

/**
 * @Author: liuxindong
 * @Description: 操作日志action
 * @Create: 2017/10/18 10:26
 * @Modify:
 */
@RestController
@RequestMapping(value = "/system/log/operaLog")
public class SysOperaLogAction extends BaseAction<SysOperaLog> {

    @Autowired
    private SysOperaLogService sysOperaLogService;

    @Override
    protected Object baseInsert(SysOperaLog sysOperaLog) {
        return null;
    }

    @Override
    protected Object baseInsertBatch(List<SysOperaLog> list) {
        return null;
    }

    @Override
    protected Object baseUpdate(SysOperaLog sysOperaLog) {
        return null;
    }

    @Override
    @DeleteMapping
    protected Object baseRemove(String code) {
        return sysOperaLogService.baseDeleteById(code);
    }

    @Override
    @DeleteMapping("/batch")
    protected Object baseRemoveBatch(String codes) {
        return sysOperaLogService.baseDeleteBatch(codes);
    }

    @Override
    @GetMapping
    protected Object baseFindById(String code) {
        return sysOperaLogService.baseFindById(code);
    }

    @Override
    protected Object baseFindAll() {
        return null;
    }

    @Override
    protected Object baseFindListByParams(SysOperaLog sysOperaLog) {
        return null;
    }

    @Override
    @GetMapping("/page")
    protected Object baseFindByPage(int page, int limit, String startTime, String endTime, String orderBy, SysOperaLog sysOperaLog) {
        return sysOperaLogService.baseFindByPage(new PageResult<>(page,limit,startTime,endTime,orderBy,sysOperaLog));
    }

    @DeleteMapping("/batch/pre30days")
    public Object remove30Early(){
        return sysOperaLogService.doRemoveEarly30();
    }
}

File: src/main/java/cn/don9cn/blog/action/system/log/SysOperaLogAction.kt
package cn.don9cn.blog.action.system.log

import cn.don9cn.blog.support.mongo.core.PageResult
import cn.don9cn.blog.action.BaseAction
import cn.don9cn.blog.model.system.log.SysOperaLog
import cn.don9cn.blog.service.system.log.SysOperaLogService
import cn.don9cn.blog.support.action.ActionMsg
import cn.don9cn.blog.support.action.ActionMsgHandler
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(value = ["/system/log/operaLog"])
open class SysOperaLogAction : BaseAction<SysOperaLog>() {

    @Autowired
    private var sysOperaLogService: SysOperaLogService? = null

    override fun baseInsert(t: SysOperaLog): ActionMsg {
        return ActionMsgHandler.insert(sysOperaLogService!!.baseInsert(t))
    }

    override fun baseInsertBatch(list: List<SysOperaLog>): ActionMsg {
        return ActionMsgHandler.insert(sysOperaLogService!!.baseInsertBatch(list))
    }

    override fun baseUpdate(t: SysOperaLog): ActionMsg {
        return ActionMsgHandler.update(sysOperaLogService!!.baseUpdate(t))
    }

    @DeleteMapping
    override fun baseRemove(code: String): ActionMsg {
        return ActionMsgHandler.delete(sysOperaLogService!!.baseDeleteById(code))
    }

    @DeleteMapping("/batch")
    override fun baseRemoveBatch(codes: String): ActionMsg {
        return ActionMsgHandler.delete(sysOperaLogService!!.baseDeleteBatch(codes))
    }

    @GetMapping
    override fun baseFindById(code: String): ActionMsg {
        return ActionMsgHandler.find(sysOperaLogService!!.baseFindById(code))
    }

    override fun baseFindAll(): ActionMsg {
        return ActionMsgHandler.find(sysOperaLogService!!.baseFindAll())
    }

    override fun baseFindListByParams(t: SysOperaLog): ActionMsg {
        return ActionMsgHandler.find(sysOperaLogService!!.baseFindListByParams(t))
    }

    @GetMapping("/page")
    override fun baseFindByPage(page: Int, limit: Int, startTime: String?, endTime: String?, orderBy: String?, t: SysOperaLog): ActionMsg {
        return ActionMsgHandler.find(sysOperaLogService!!.baseFindByPage(PageResult(page, limit, startTime, endTime, orderBy, t)))
    }

    @DeleteMapping("/batch/pre30days")
    open fun remove30Early(): ActionMsg {
        return ActionMsgHandler.delete(sysOperaLogService!!.doRemoveEarlyDays(30))
    }
}